File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F17.P1.swift
import Foundation

// BEGIN PROBE F17.P1
func readAll() throws -> Int {
    let handle = try FileHandle(forReadingFrom: URL(filePath: "data.txt"))
    defer { try? handle.close() }
    let text = String(decoding: try handle.readToEnd() ?? Data(), as: UTF8.self)
    return text.utf8.count
}
// END PROBE F17.P1

print(try readAll())

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F17.P1.go
package main

import (
	"fmt"
	"io"
	"os"
)

// BEGIN PROBE F17.P1
func readAll() (int, error) {
	f, err := os.Open("data.txt")
	if err != nil {
		return 0, err
	}
	defer f.Close()
	b, err := io.ReadAll(f)
	if err != nil {
		return 0, err
	}
	text := string(b)
	return len(text), nil
}

// END PROBE F17.P1

func main() {
	n, err := readAll()
	if err != nil {
		panic(err)
	}
	fmt.Println(n)
}
